packages/cli-link-assets/src/linkAssets.ts

```diff
@@ -0,0 +1,227 @@
+import type {Config as CLIConfig} from '@react-native-community/cli-types';
+import fs from 'fs';
+import path from 'path';
+import {audioTypes, fontTypes, imageTypes} from './fileTypes';
+import cleanAndroidAssets from './tools/cleanAssets/android';
+import cleanIOSAssets from './tools/cleanAssets/ios';
+import copyAndroidAssets from './tools/copyAssets/android';
+import copyIOSAssets from './tools/copyAssets/ios';
+import linkPlatform, {
+  LinkOptions,
+  LinkOptionsPerExt,
+  LinkPlatformOptions,
+} from './tools/linkPlatform';
+import getManifest from './tools/manifest';
+
+type Args = {
+  projectRoot: string;
+};
+
+function findPbxprojFile(files: Array<string>): string | null {
+  const sortedFiles = files.sort();
+
+  for (let i = sortedFiles.length - 1; i >= 0; i--) {
+    const fileName = files[i];
+    const ext = path.extname(fileName);
+
+    if (ext === '.pbxproj') {
+      return fileName;
+    }
+  }
+
+  return null;
+}
```

Comment:
`findXcodeProject` doesn't give me the `.pbxproj` file, so we would have to adjust it to return it together with the `IOSProjectInfo` object. WDYT?

Reply:
or maybe let's add `path` to return it from `findXcodeProject` and then create a function that takes `IOSProjectInfo` as param and just concatenate path, filename and `.pbxproj`? IIRC `.pbxproj` is a required file so this should do the work. Then this function could live in `cli-platform-apple`

Reply:
Not sure if I understood correctly, I tried this approach for example:

```ts
import {IOSProjectInfo} from '@react-native-community/cli-types';
import path from 'path';

function findPbxprojFile(projectInfo: IOSProjectInfo): string {
  return path.join(projectInfo.path, projectInfo.name, 'project.pbxproj');
}

export default findPbxprojFile;
```

In `findXcodeProject` I'm returning a new `path` property that is basically `path.dirname(fileName);`:

```ts
function findXcodeProject(files: Array<string>): IOSProjectInfo | null {
  const sortedFiles = files.sort();

  for (let i = sortedFiles.length - 1; i >= 0; i--) {
    const fileName = files[i];
    const ext = path.extname(fileName);
    const projectPath = path.dirname(fileName);

    if (ext === '.xcworkspace') {
      return {
        name: fileName,
        path: projectPath,
        isWorkspace: true,
      };
    }
    if (ext === '.xcodeproj') {
      return {
        name: fileName,
        path: projectPath,
        isWorkspace: false,
      };
    }
  }

  return null;
}
```

The issue is, `project.pbxproj` is always under `.xcodeproj` folder, so if I concatenate `path`, `name` and `project.pbxproj` and `name` is e.g. `AwesomeApp.xcworkspace` because we do have a workspace, the logic won't work well.

Reply:
after internal discussion, let's just replace `.xcworkspace` in `findPbxprojFile` `projectInfo.name`. They are always on the same level so it should be safe